pkg/roll/update_test.go

```diff
@@ -683,8 +684,81 @@ func TestShouldRollMidwayHealthyMigrationFromZeroWhenNewSatisfies(t *testing.T)
 	upd.DesiredReplicas = 3
 	upd.MinimumReplicas = 2
 
-	remove, add, err := upd.shouldRollAfterDelay(rc_fields.RC{ID: upd.NewRC, Manifest: manifest})
+	remove, add, err := upd.shouldRollAfterDelay(manifest.ID())
 	Assert(t).IsNil(err, "expected no error determining nodes to roll")
 	Assert(t).AreEqual(remove, 0, "expected to remove no nodes")
 	Assert(t).AreEqual(add, 1, "expected to add one node")
 }
+
+func failOnError(t *testing.T, desc string, errs <-chan error) {
+	if err, ok := <-errs; ok {
+		t.Fatalf("Error received on " + desc + ":" + err.Error())
+	}
+}
+
+// Transfers the named node from the old RC to the new RC
+func transferNode(node string, manifest pods.Manifest, upd update) error {
+	if _, err := upd.kps.SetPod(kp.REALITY_TREE, node, manifest); err != nil {
+		return err
+	}
+	podID := manifest.ID().String()
+	return upd.labeler.SetLabel(labels.POD, node+"/"+podID, rc.RCIDLabel, string(upd.NewRC))
+}
+
+func TestRollLoopTypicalCase(t *testing.T) {
+	upd, _, manifest := updateWithHealth(t, 3, 0, nil, nil, nil)
+	upd.DesiredReplicas = 3
+	upd.MinimumReplicas = 2
+
+	healths := make(chan map[string]health.Result)
+
+	oldRC := rc_fields.RC{ID: upd.OldRC}
+	oldRCUpdated, oldRCErrors := upd.rcs.Watch(&oldRC, nil)
+	go failOnError(t, "old RC", oldRCErrors)
+
+	newRC := rc_fields.RC{ID: upd.NewRC}
+	newRCUpdated, newRCErrors := upd.rcs.Watch(&newRC, nil)
+	go failOnError(t, "new RC", newRCErrors)
+
+	rollLoopResult := make(chan bool)
+
+	go func() {
+		rollLoopResult <- upd.rollLoop(manifest.ID(), healths, nil, nil)
+		close(rollLoopResult)
+	}()
+
+	checks := map[string]health.Result{
+		"node1": {Status: health.Passing},
+		"node2": {Status: health.Passing},
+		"node3": {Status: health.Passing},
+	}
+
+	healths <- checks
+
+	<-oldRCUpdated
```

Comment:
helper function to wait on both `oldRCUpdated` and `newRCUpdated` and assert on their values? This code got a bit repetitive.